### PR TITLE
GG-22620 Support DR alerts.

### PR DESCRIPTION
**Pull request by akuznetsov-gridgain**

### CODE REVIEW

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/db/OneToManyIndex.java [hunk 1]
```diff
@@ -74,10 +75,21 @@ public OneToManyIndex(Ignite ignite, String idxName) {
...
 
+    /**
+     * @param parentIds Parent IDs.
+     * @return Set of children IDs.
+     */
+    public Set<V> getAll(Set<K> parentIds) {
+        return parentIds
+            .stream()
+            .flatMap(accId -> get(accId).stream())
+            .collect(Collectors.toSet());
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/db/OneToManyIndex.java]:
```suggestion
            .collect(toSet());
```

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java [hunk 2]
```diff
@@ -401,4 +409,16 @@ private void updateAgentsCache(WebSocketSession ses, Set<UUID> accIds) {
...
+    /**
+     * @return List of connected account.
+     */
+    protected Set<UUID> connectedAccounts() {
+        return locAgents
+            .values()
+            .stream()
+            .filter(ses -> !F.isEmpty(ses.getClusterIds()))
+            .flatMap(ses -> ses.getAccIds().stream())
+            .collect(Collectors.toSet());
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java]:
```suggestion
            .collect(toSet());
```

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java [hunk 3]
```diff
@@ -401,4 +409,16 @@ private void updateAgentsCache(WebSocketSession ses, Set<UUID> accIds) {
 
         locAgents.put(ses, new AgentSession(accIds));
     }
+
+    /**
+     * @return List of connected account.
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java]:
```suggestion
     * @return Locally connected accounts.
```

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/db/OneToManyIndex.java (hunk 1 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/db/OneToManyIndex.java]: Fixed

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java (hunk 3 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java]: I removed this method.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java (hunk 2 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java]: I removed this method.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/BrowsersService.java [hunk 4]
```diff
@@ -359,17 +358,18 @@ protected JsonObject fillVisorGatewayTaskParams(JsonObject payload) {
...
+     * @return {@code true} if response was processed.
      */
-    protected void sendResponseToBrowser(WebSocketEvent evt) {
+    protected boolean processResponse(WebSocketEvent evt) {
         WebSocketSession ses = locRequests.remove(evt.getRequestId());
 
-        if (ses == null) {
-            log.warn("Failed to send event to browser: " + evt);
+        if (ses != null) {
+            sendMessageQuiet(ses, evt);
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/BrowsersService.java]: Looks like we should catch error if we need return correct flag

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/ClustersRepository.java [hunk 5]
```diff
@@ -171,32 +183,33 @@ public boolean hasDemo(UUID accId) {
...
+            Set<String> clusterIds = stream(clusters.cache().spliterator(), false)
                 .filter(entry -> entry.getValue().isExpired(DEFAULT_CLUSTER_CLEANUP))
                 .map(Cache.Entry::getKey)
                 .collect(toSet());
 
             if (!F.isEmpty(clusterIds)) {
                 clusters.cache().removeAll(clusterIds);
 
-                log.debug("Failed to receive topology update for clusters: " + clusterIds);
+                if (log.isDebugEnabled())
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/ClustersRepository.java]: Why we should check log level for slf4j logger?

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java [hunk 6]
```diff
@@ -162,7 +164,13 @@ private void registerListeners() {
 
         ignite.message().localListen(SEND_RESPONSE, new MessagingListenActor<WebSocketEvent>() {
             @Override protected void receive(UUID nodeId, WebSocketEvent evt) {
-                browsersSrvc.sendResponseToBrowser(evt);
+                boolean processed = browsersSrvc.processResponse(evt);
+
+                if (!processed)
+                    processed = agentsSrvc.processResponse(evt);
+
+                if (!processed)
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java]: Why we again check this condition?

Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java]: We check condition to show warning that response was not processed.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/ClustersRepository.java (hunk 5 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/ClustersRepository.java]: Fixed

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java [hunk 7]
```diff
@@ -145,7 +151,10 @@ public AgentsService(Ignite ignite, AccountsRepository accRepo, AgentsRepository
                 try {
                     UUID nid = srcOfRequests.remove(evt.getRequestId());
 
-                    ignite.message(ignite.cluster().forNodeId(nid)).send(SEND_RESPONSE, evt);
+                    if (nid != null)
+                        transitionSrvc.sendResponse(nid, evt);
+                    else
+                        log.warn("Request ID not found for response: " + evt);
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java]: Detected response with duplicated or unexpected id

**Pull request merged**